app/src/main/java/com/example/campusguide/search/CustomSearch.kt

```diff
@@ -0,0 +1,55 @@
+package com.example.campusguide.search
+
+import android.app.Activity
+import android.content.Intent
+import android.view.View
+import com.example.campusguide.ActivityResultListener
+import com.example.campusguide.Constants
+import com.example.campusguide.MapsActivity
+import com.example.campusguide.map.Map
+import com.example.campusguide.utils.DisplayMessageErrorListener
+import com.google.android.libraries.places.api.Places
+import com.google.android.libraries.places.api.model.Place
+import com.google.android.libraries.places.api.net.FetchPlaceRequest
+
+const val AUTOCOMPLETE_REQUEST_CODE = 69; //nice
+
+class CustomSearch constructor(private val activity: MapsActivity, private val map: Map) : View.OnClickListener,
+    ActivityResultListener {
+    override fun onClick(v: View?) {
+        val searchIntent = Intent(activity, CustomSearchActivity::class.java)
+        activity.startActivityForResult(searchIntent, AUTOCOMPLETE_REQUEST_CODE)
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (requestCode != AUTOCOMPLETE_REQUEST_CODE || resultCode != Activity.RESULT_OK) {
+            return
+        }
+
+        val result = data?.data?.toString()
+        if (result != null) {
+            getPlace(result)
+        }
```

Comment:
Should something be done when result is null?

Reply:
At this point the activity has returned status OK, so I don't see any way that it could be null here. I decided to silently fail rather than clutter the code, & we can add an error popup if it ends up causing problems